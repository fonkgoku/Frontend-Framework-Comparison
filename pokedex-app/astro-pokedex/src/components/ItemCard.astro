---
const { postId, height, weight, types, name } = Astro.props;
---

<div class="tile">
  <figure class="image">
    <img
      src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${postId}.png`}
      alt={name}
    />
  </figure>
  <div class="content">
    <h3>
      <span class="prefix">#{postId}:&nbsp;</span>
      {name}
    </h3>
    {
      height && weight && types && (
        <table class="mb-4 mt-4">
          <tbody>
            <tr>
              <th>Height:</th>
              <td>{height}</td>
            </tr>
            <tr>
              <th>Weight:</th>
              <td>{weight}</td>
            </tr>
            <tr>
              <th>Types:</th>
              <td>{types.map((type) => type.type.name).join(", ")}</td>
            </tr>
          </tbody>
        </table>
      )
    }
  </div>
</div>

<style></style>
