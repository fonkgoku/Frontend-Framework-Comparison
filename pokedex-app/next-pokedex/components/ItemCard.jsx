function ItemCard({ postId, name, height, weight, types }) {
  return (
    <div className="tile">
      <figure className="image">
        <img
          src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${postId}.png`}
          alt={name}
        />
      </figure>
      <div className="content">
        <h3>
          <span className="prefix">#{postId}:&nbsp;</span>
          {name}
        </h3>

        {height && weight && types && (
          <table className="mb-4 mt-4">
            <tbody>
              <tr>
                <th>Height:</th>
                <td>{height}</td>
              </tr>
              <tr>
                <th>Weight:</th>
                <td>{weight}</td>
              </tr>
              <tr>
                <th>Types:</th>
                <td>{types.map((type) => type.type.name).join(', ')}</td>
              </tr>
            </tbody>
          </table>
        )}
      </div>
    </div>
  )
}

export default ItemCard
