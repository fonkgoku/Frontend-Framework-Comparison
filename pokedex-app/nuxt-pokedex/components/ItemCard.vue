<template>
  <div class="tile">
    <figure class="image">
      <img
        :src="`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${postId}.png`"
        :alt="name"
      />
    </figure>
    <div class="content">
      <h3>
        <span class="prefix">#{{ postId }}:&nbsp;</span>
        {{ name }}
      </h3>
    </div>
    <template v-if="height && weight && types">
      <DetailTable :height="height" :weight="weight" :types="types" />
    </template>
  </div>
</template>

<script>
import DetailTable from './DetailTable.vue'
export default {
  props: {
    postId: Number,
    name: String,
    height: Number,
    weight: Number,
    types: Array,
  },
  components: { DetailTable },
}
</script>
