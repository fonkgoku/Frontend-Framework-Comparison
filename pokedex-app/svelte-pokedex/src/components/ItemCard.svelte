<script>
    import ItemCard from './ItemCard.svelte';
	export let postId;
	export let name;
</script>

<div class="tile">
	<figure class="image">
		<img
			src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${postId}.png`}
			alt={name}
		/>
	</figure>
	<div class="content">
		<h3>
			<span class="prefix">#{postId}:&nbsp;</span>
			{name}
		</h3>
        <ItemCard {...$$restProps} />
	</div>
</div>
