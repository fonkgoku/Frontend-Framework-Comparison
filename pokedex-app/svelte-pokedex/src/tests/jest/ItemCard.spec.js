import { render, screen } from '@testing-library/svelte';
import ItemCard from '@/components/ItemCard.svelte';

const ITEM_DATA = {
	postId: 1,
	name: 'Bulbasaur',
	height: 100,
	weight: 100,
	types: [{ type: { name: 'grass' } }]
};

test('ItemCard: shows proper heading when rendered', () => {
	render(ItemCard);
	// const heading = screen.getByRole('heading', {
	// 	name: `#${ITEM_DATA.postId}: ${ITEM_DATA.name}`
	// });
	const node = screen.queryByText("#");
	// expect(heading).not.toBeNull();
	expect(node).not.toBeNull();
});
