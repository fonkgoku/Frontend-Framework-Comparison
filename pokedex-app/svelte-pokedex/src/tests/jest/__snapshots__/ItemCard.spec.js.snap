// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard renders correctly 1`] = `
<body>
  <div>
    <div
      class="tile"
    >
      <figure
        class="image"
      >
        <img
          alt="Bulbasaur"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
        />
      </figure>
       
      <div
        class="content"
      >
        <h3>
          <span
            class="prefix"
          >
            #
            1
            :Â 
          </span>
           
          Bulbasaur
        </h3>
         
        <table
          class="mb-4 mt-4"
        >
          <tbody>
            <tr>
              <th>
                Height:
              </th>
               
              <td>
                100
              </td>
            </tr>
             
            <tr>
              <th>
                Weight:
              </th>
               
              <td>
                100
              </td>
            </tr>
             
            <tr>
              <th>
                Types:
              </th>
               
              <td>
                grass
              </td>
            </tr>
          </tbody>
        </table>
        
      </div>
    </div>
  </div>
</body>
`;
